ignite/templates/app/files/Makefile.plush

```diff
@@ -0,0 +1,78 @@
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+COMMIT := $(shell git log -1 --format='%H')
+APPNAME := <%= AppName %>
+
+# don't override user values
+ifeq (,$(VERSION))
+  VERSION := $(shell git describe --exact-match 2>/dev/null)
+  # if VERSION is empty, then populate it with branch's name and raw commit hash
+  ifeq (,$(VERSION))
+    VERSION := $(BRANCH)-$(COMMIT)
+  endif
+endif
+
+# Update the ldflags with the app, client & server names
+ldflags = -X github.com/cosmos/cosmos-sdk/version.Name=$(APPNAME) \
+	-X github.com/cosmos/cosmos-sdk/version.AppName=$(APPNAME)d \
+	-X github.com/cosmos/cosmos-sdk/version.Version=$(VERSION) \
+	-X github.com/cosmos/cosmos-sdk/version.Commit=$(COMMIT)
+
+BUILD_FLAGS := -ldflags '$(ldflags)'
+
+#################
+###   Build   ###
+#################
+
+test:
+	@echo "--> Running tests"
+	go test -v ./...
+
+.PHONY: test
+
+#################
+###  Install  ###
+#################
+
+all: install
+
+install:
+	@echo "--> ensure dependencies have not been modified"
+	@go mod verify
+	@echo "--> installing $(APPNAME)d"
+	@go install $(BUILD_FLAGS) -mod=readonly ./cmd/$(APPNAME)d
+
+.PHONY: all install
+
+##################
+###  Protobuf  ###
+##################
+
+# Use this proto-image if you do not want to use Ignite for generating proto files
+protoVer=0.15.1
+protoImageName=ghcr.io/cosmos/proto-builder:$(protoVer)
+protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName)
+
+proto-gen:
+	@echo "Generating protobuf files..."
+	@ignite generate proto-go --yes
+
+.PHONY: proto-gen
+
+#################
+###  Linting  ###
+#################
+
+golangci_lint_cmd=golangci-lint
+golangci_version=v1.61.0
+
+lint:
+	@echo "--> Running linter"
+	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(golangci_version)
+	@$(golangci_lint_cmd) run ./... --timeout 15m
+
+lint-fix:
+	@echo "--> Running linter and fixing issues"
+	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(golangci_version)
+	@$(golangci_lint_cmd) run ./... --fix --timeout 15m
+
+.PHONY: lint lint-fix
```

Comment:
Do you think we should add the `test-sim-*` methods?

Reply:
```suggestion
.PHONY: lint lint-fix

###############################################################################
###                               Development                               ###
###############################################################################

## govet: Run go vet.
govet:
	@echo Running go vet...
	@go vet ./...

## govulncheck: Run govulncheck
govulncheck:
	@echo Running govulncheck...
	@go run golang.org/x/vuln/cmd/govulncheck ./...

FIND_ARGS := -name '*.go' -type f -not -name '*.pb.go' -not -name '*.pb.gw.go' -not -path './api/*'

## format: Run gofumpt and goimports.
format:
	@echo Formatting...
	@go install mvdan.cc/gofumpt
	@go install golang.org/x/tools/cmd/goimports
	@find . $(FIND_ARGS) | xargs gofumpt -w .
	@find . $(FIND_ARGS) | xargs goimports -w -local github.com/ignite/network

.PHONY: format govet govulncheck


###############################################################################
###                                  Test                                   ###
###############################################################################

## test-unit: Run the unit tests.
test-unit:
	@echo Running unit tests...
	@VERSION=$(VERSION) go test -mod=readonly -v -timeout 30m $(PACKAGES)

## test-race: Run the unit tests checking for race conditions
test-race:
	@echo Running unit tests with race condition reporting...
	@VERSION=$(VERSION) go test -mod=readonly -v -race -timeout 30m  $(PACKAGES)

## test-cover: Run the unit tests and create a coverage html report
test-cover:
	@echo Running unit tests and creating coverage report...
	@VERSION=$(VERSION) go test -mod=readonly -v -timeout 30m -coverprofile=$(COVER_FILE) -covermode=atomic $(PACKAGES)
	@go tool cover -html=$(COVER_FILE) -o $(COVER_HTML_FILE)
	@rm $(COVER_FILE)

## bench: Run the unit tests with benchmarking enabled
bench:
	@echo Running unit tests with benchmarking...
	@VERSION=$(VERSION) go test -mod=readonly -v -timeout 30m -bench=. $(PACKAGES)

## test: Run unit and integration tests.
test: govet govulncheck test-unit

.PHONY: test test-unit test-race test-cover bench

help: Makefile
	@echo
	@echo " Choose a command run in "$(APPNAME)", or just run 'make' for install"
	@echo
	@sed -n 's/^##//p' $< | column -t -s ':' |  sed -e 's/^/ /'
	@echo

.PHONY: help
```

Reply:
Yes, that's good ones 👌🏾